test/units/module_utils/common/collections.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+"""Test low-level utility functions from ``module_utils.common.collections``."""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from collections.abc import Sequence
+
+import pytest
+
+from ansible.module_utils.common.collections import is_iterable, is_sequence
+
+
+class SeqStub:
+    """Stub emulating a sequence type.
+
+    >>> from collections.abc import Sequence
+    >>> assert issubclass(SeqStub, Sequence)
+    >>> assert isinstance(SeqStub(), Sequence)
+    """
+
+
+Sequence.register(SeqStub)
+
+
+class IteratorStub:
+    def __next__(self):
+        raise StopIteration
+
+
+class IterableStub:
+    def __iter__(self):
+        return IteratorStub()
+
+
+TEST_STRINGS = 'he', b'he', u'Україна', u'Česká republika'
```

Comment:
For what we're testing, I would make sure we have the same data in both text and byte string form.  For bytes, we currently only need to test utf-8 encoded test:
``` python
TEST_STRINGS = (u'he', u'Україна', u'Česká republika')
TEST_STRINGS = TEST_STRINGS + tuple(s.encode('utf-8') for s in TEST_STRINGS)
```